src/schema/submissions.ts

```diff
@@ -0,0 +1,113 @@
+import { Post, Submission, User } from './../entity';
+import { IResolvers } from 'graphql-tools';
+import { traceResolvers } from './trace';
+import { Context } from '../Context';
+import { getDiscussionLink, isValidHttpUrl } from '../common';
+import { ValidationError } from 'apollo-server-errors';
+import { GQLEmptyResponse } from './common';
+
+interface GQLArticleSubmission {
+  url: string;
+}
+
+interface GQLSubmissionAvailability {
+  hasAccess: boolean;
+  limit: number;
+  todaySubmissionsCount: number;
+}
+
+export const DEFAULT_SUBMISSION_LIMIT = '3';
+
+export const typeDefs = /* GraphQL */ `
+  type SubmissionAvailability {
+    hasAccess: Boolean!
+    limit: Int!
+    todaySubmissionsCount: Int!
+  }
+
+  extend type Query {
+    """
+    Information regarding the access of submitting community links
+    """
+    submissionsAndLimit: SubmissionAvailability!
+  }
+  extend type Mutation {
+    """
+    Submit an article to surface on users feed
+    """
+    submitArticle(url: String!): EmptyResponse @auth
+  }
+`;
+
+export const resolvers: IResolvers<unknown, Context> = traceResolvers({
+  Query: {
+    submissionsAndLimit: async (
+      _,
+      __,
+      ctx,
+    ): Promise<GQLSubmissionAvailability> => {
+      const limit = parseInt(
+        process.env.SCOUT_SUBMISSION_LIMIT || DEFAULT_SUBMISSION_LIMIT,
+      );
+
+      const repo = ctx.getRepository(Submission);
+      const user = await ctx.getRepository(User).findOne({ id: ctx.userId });
+      if (!user) {
+        return { limit, hasAccess: false, todaySubmissionsCount: 0 };
+      }
+
+      const timezone = user.timezone ?? 'utc';
+      const nowTimezone = `timezone('${timezone}', now())`;
+      const atTimezone = `at time zone '${timezone}'`;
+      const submissions = await repo.find({
+        where: `date_trunc('day', "createdAt")::timestamptz ${atTimezone} = date_trunc('day', ${nowTimezone})::timestamptz`,
+      });
+      const hasAccess = true; // subject for change on how we define the access
+
+      return { limit, todaySubmissionsCount: submissions.length, hasAccess };
+    },
+  },
+  Mutation: {
+    submitArticle: async (
+      _,
+      { url }: GQLArticleSubmission,
+      ctx,
+    ): Promise<GQLEmptyResponse> => {
+      if (!isValidHttpUrl(url)) {
+        throw new ValidationError('Invalid URL!');
+      }
+
+      const postRepo = ctx.con.getRepository(Post);
+      const existingPost = await postRepo
+        .createQueryBuilder('post')
+        .where('url = :url or "canonicalUrl" = :url', { url })
+        .andWhere('deleted = false')
```

Comment:
This is some what tricky. Because if there's a deleted post with the same url or canonical url you will not be able to add it to the db

Reply:
Should we remove the deleted condition so it won't have to go through the process before realizing it existed and deleted?

Reply:
Changed the way we handle deleted posts :)